dwarf.go

```diff
@@ -308,20 +308,34 @@ func (d *dwarfmapper) Locations(fn experimental.InternalFunction, pc experimenta
 		File:       le.File.Name,
 		Line:       int64(le.Line),
 		Column:     int64(le.Column),
-		Inlined:    len(spgm.Inlines) > 0,
+		Inlined:    false,
 		HumanName:  human,
 		StableName: stable,
 	})
 
 	if len(spgm.Inlines) > 0 {
 		files := lr.Files()
 		for i := len(spgm.Inlines) - 1; i >= 0; i-- {
-			// TODO: check source offset is in range of inline?
 			f := spgm.Inlines[i]
 			fileIdx, ok := f.Val(dwarf.AttrCallFile).(int64)
 			if !ok || fileIdx >= int64(len(files)) {
 				break
 			}
+
+			ranges, err := d.d.Ranges(f)
+			if err != nil {
+				continue
+			}
+			found := false
+			for _, x := range ranges {
+				if x[0] <= offset && offset <= x[1] {
+					found = true
+				}
+			}
+			if !found {
+				continue
+			}
```

Comment:
Whenever I write this kind of control flow I usually choose to move the loop into a function returning a boolean so I can get rid of the boolean value and have code that looks like this:
```go
if !rangesContainOffset(ranges, offset) {
    continue
}
```

Reply:
Pushed on this branch a commit I was working on in parallel. I started moving some of the dwarf information gathering to when the dwarf section is initially parsed. As a result, it does fix this feedback. I'll push the rest in a separate PR.